### ISSUE TITLE
[Feature Request] A setting to avoid the window "Download linked media"

### ISSUE THREAD

**Issue by jdbonjour** [activityworkshop/GpsPrune#103]
When loading a GPX file containing media information (http...), a window "Download linked media" appears, and we need to either click [Block] or [Allow] for each media ! So it would be useful to have a configuration parameter (BlockAll or AllowAll) to avoid this window

**Comment by activityworkshop** [comment]
It shouldn't ask you this "for each media", it should only ask once per domain.  The expectation was that a single file probably refers to media on the same domain, so the number of questions should be small.  And an "Allow All" button would prevent you from seeing what the domains are.

Maybe if there's a large number of different domains used by a single file then it should list them all in advance and just ask you once to "allow all these domains" or "block all these domains"?  That would definitely need fewer clicks than allowing or blocking each one by one.

**Comment by jdbonjour** [comment]
Yes, a solution oriented "allow all these domains" or "block all these domains" would be a good step in the right direction! Thanks very much in advance !

**Comment by jdbonjour** [comment]
Mail précédent : c'est une réponse à ma question concernant GpsPrune posée
sur GitHub :

When loading a GPX file containing media information (http...), a window
"Download linked media" appears, and we need to either click [Block] or
[Allow] for each media ! So it would be useful to have a configuration
parameter (BlockAll or AllowAll) to avoid this window.

Je répond ensuite :

Yes, a solution oriented "allow all these domains" or "block all these
domains" would be a good step in the right direction! Thanks very much in
advance !

**Comment by activityworkshop** [comment]
Here is an early look at how this dialog might present itself. It's not _exactly_ what you asked for, but it goes in this direction.

Firstly, if you load a gpx file with only one or two photos in it, then the behaviour is unchanged - you will get the existing popup for the first photo, and if the second photo is on a different domain then you will get a popup for that second domain too. You can choose to "Allow" or "Block" for the one or two domains as before.

Secondly, if you load a gpx file with a hundred photos but they all come from one or two domains, then that is also unchanged - just click "Allow" or "Block" one or two times.

Now, it gets different when you load a gpx file linking to photos coming from three or more domains, then you'll in future get the following dialog asking what you want to do.

![Image](https://github.com/user-attachments/assets/936035e4-07e8-4b67-a2fa-6bfef9281b08)

You can see all the domains but not all the filenames, and you get three choices:
 * Allow all, means photos from all these listed domains will be allowed, and no more popups for them.
 * Block all, means photos from all these listed domains will be blocked, and no more popups for them.
 * Choose, means I'm not sure and I want to go through them one by one with the individual popups like before.
 
This is on a per-file basis, so after you've done that for your first gpx file, and then load a second gpx file, the same rules apply to the domains used in that one too - the ones you have already allowed are allowed, the ones you have already blocked are blocked, but it will present choices for the domains which haven't appeared before.

I'm not sure if the way the domains are presented in a list is confusing or not - some users might think that they can use this list to just select the first two of the four, but that's not the way it currently works. You can't select any of the domains from this list, and your only choices are to allow all, to block them all, or do nothing and wait for the following popups which will also show the full URLs as before.

This is planned for version 26, but feedback is welcome.
